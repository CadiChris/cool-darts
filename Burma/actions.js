import Contrats from "./Contrats";

export const INSCRIRE_JOUEUR = "INSCRIRE_JOUEUR";
export const inscrireJoueur = nomDuJoueur => ({
  type: INSCRIRE_JOUEUR,
  payload: {
    nomDuJoueur
  }
});

export const DEMARRER_PARTIE = "DEMARRER_PARTIE";
export const demarrerPartie = () => ({
  type: DEMARRER_PARTIE
});

export const VOLEE = "VOLEE";
export const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.chiffre(chiffre, nombreDeTouches)
  }
});

export const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.double(chiffresTouches)
  }
});

export const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.triple(chiffresTouches)
  }
});

export const voleeSurBull = (
  lanceur,
  nombreDeSimplesBull,
  nombreDeDoublesBull
) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.bull(nombreDeSimplesBull, nombreDeDoublesBull)
  }
});
