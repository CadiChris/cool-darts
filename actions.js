const INSCRIRE_JOUEUR = "INSCRIRE_JOUEUR";
const inscrireJoueur = nomDuJoueur => ({
  type: INSCRIRE_JOUEUR,
  payload: {
    nomDuJoueur
  }
});

const DEMARRER_PARTIE = "DEMARRER_PARTIE";
const demarrerPartie = () => ({
  type: DEMARRER_PARTIE
});

const VOLEE_SUR_CHIFFRE = "VOLEE_SUR_CHIFFRE";
const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE_SUR_CHIFFRE,
  payload: {
    lanceur,
    chiffre,
    nombreDeTouches
  }
});

const VOLEE_SUR_DOUBLE = "VOLEE_SUR_DOUBLE";
const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE_SUR_DOUBLE,
  payload: {
    lanceur,
    chiffresTouches
  }
});

const VOLEE_SUR_TRIPLE = "VOLEE_SUR_TRIPLE";
const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE_SUR_TRIPLE,
  payload: {
    lanceur,
    chiffresTouches
  }
});

const VOLEE_SUR_BULL = "VOLEE_SUR_BULL";
const voleeSurBull = (lanceur, nombreDeSimplesBull, nombreDeDoublesBull) => ({
  type: VOLEE_SUR_BULL,
  payload: {
    lanceur,
    nombreDeSimplesBull,
    nombreDeDoublesBull
  }
});

export {
  INSCRIRE_JOUEUR,
  inscrireJoueur,
  DEMARRER_PARTIE,
  demarrerPartie,
  VOLEE_SUR_CHIFFRE,
  VOLEE_SUR_DOUBLE,
  VOLEE_SUR_TRIPLE,
  VOLEE_SUR_BULL,
  voleeSurChiffre,
  voleeSurDouble,
  voleeSurTriple,
  voleeSurBull
};
