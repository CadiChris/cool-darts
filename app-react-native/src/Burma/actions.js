import Contrats from "./arbitrage/Contrats";

export const DEMARRER_PARTIE = "DEMARRER_PARTIE";
export const demarrerPartie = joueurs => ({
  type: DEMARRER_PARTIE,
  joueurs
});

export const VOLEE = "VOLEE";
export const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.chiffre(chiffre, nombreDeTouches)
  }
});

export const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.double(chiffresTouches)
  }
});

export const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.triple(chiffresTouches)
  }
});

export const voleeSurBull = (lanceur, nombreDeBulls) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.bull(nombreDeBulls)
  }
});
