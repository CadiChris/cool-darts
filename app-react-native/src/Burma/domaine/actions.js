import Contrats from './arbitrage/Contrats';

export const DEMARRER_BURMA = 'BURMA/DEMARRER';
export const demarrerBurma = joueurs => ({
  type: DEMARRER_BURMA,
  joueurs,
});

export const VOLEE = 'BURMA/VOLEE';
export const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.chiffre(chiffre, nombreDeTouches),
  },
});

export const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.double(chiffresTouches),
  },
});

export const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.triple(chiffresTouches),
  },
});
