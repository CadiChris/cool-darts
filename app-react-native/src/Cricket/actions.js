export const demarrerPartie = joueurs => ({
  type: "DEMARRER_PARTIE",
  joueurs
});

export const lancerFlechette = (joueur, chiffre, touches) => ({
  type: "LANCER_FLECHETTE",
  joueur,
  chiffre,
  touches
});

export const nouvellePartie = () => ({
  type: "NOUVELLE_PARTIE"
});
