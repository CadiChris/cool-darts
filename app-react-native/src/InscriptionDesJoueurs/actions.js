export const INSCRIRE_JOUEUR = "INSCRIRE_JOUEUR";
export const inscrireJoueur = nomDuJoueur => ({
  type: INSCRIRE_JOUEUR,
  nomDuJoueur
});

export const DESINSCRIRE_JOUEUR = "DESINSCRIRE_JOUEUR";
export const desinscrireJoueur = nomDuJouer => ({
  type: DESINSCRIRE_JOUEUR,
  nomDuJouer
});

export const CHOISIR_JEU = "CHOISIR_JEU";
export const choisirJeu = jeuChoisi => ({
  type: CHOISIR_JEU,
  jeuChoisi
});
