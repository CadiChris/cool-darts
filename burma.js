import { INSCRIRE_JOUEUR, DEMARRER_PARTIE, VOLEE } from "./burma.actions";
import * as lanceur from "./lanceur";
import { scores } from "./scores";
import { leChiffreSuivant } from "./chiffre";

const burma = (state = STATE_INITIAL, action) => {
  switch (action.type) {
    case INSCRIRE_JOUEUR:
      return {
        ...state,
        joueurs: [...state.joueurs, action.payload.nomDuJoueur],
        scores: scores(state.scores, action)
      };

    case DEMARRER_PARTIE:
      return {
        ...state,
        lanceur: state.joueurs[0],
        scores: scores(state.scores, action),
        phase: "EN_COURS",
        chiffreCourant: 15
      };

    case VOLEE:
      return {
        ...state,
        lanceur: lanceur.suivant(state.joueurs, action.payload.lanceur),
        chiffreCourant: leChiffreSuivant(state.chiffreCourant).avec(
          state.joueurs,
          action.payload.lanceur
        ),
        scores: scores(state.scores, action),
        phase: derniereVolee(
          state.joueurs,
          action.payload.lanceur,
          action.payload.chiffre
        )
          ? "TERMINEE"
          : state.phase
      };

    default:
      return state;
  }
};

const derniereVolee = (tousLesJoueurs, lanceur, chiffre) =>
  tousLesJoueurs.indexOf(lanceur) === tousLesJoueurs.length - 1 &&
  chiffre === "B";

const STATE_INITIAL = {
  joueurs: [],
  scores: scores(undefined, {}),
  phase: "INSCRIPTION",
  lanceur: undefined,
  chiffreCourant: undefined
};

export { burma };
