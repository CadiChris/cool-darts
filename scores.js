import { VOLEE, INSCRIRE_JOUEUR } from "./burma.actions";

function scores(state = [], action) {
  switch (action.type) {

    case  INSCRIRE_JOUEUR :
      return [
        ...state,
        scoreVierge(action.payload.nomDuJoueur)
      ]

    case VOLEE:
      return state.map((s) =>
          noterUneVolee(action.payload.chiffre, action.payload.nombreDeTouches, s, action.payload.lanceur))

    default:
      return state
  }
}

const scoreVierge = (joueur) => ({
  joueur,
  points: POINTS_INITIAUX,
  touches: {}
})
const POINTS_INITIAUX = 40;

const noterUneVolee = (chiffre, nombreDeTouches, score, lanceur) => {
  const joueurNonConcerne = score.joueur !== lanceur
  if (joueurNonConcerne)
    return { ...score }

  return {
    ...score,
    points: nombreDeTouches !== 0
        ? score.points + chiffre * nombreDeTouches
        : Math.round(score.points / 2),
    touches: {
      ...score.touches,
      [chiffre]: nombreDeTouches
    }
  }
}


export {
  scores
}