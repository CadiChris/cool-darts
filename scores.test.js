import { scores } from "./scores";
import { inscrireJoueur, voleeChiffree, voleeSurDouble } from "./burma.actions";

it("donne des points", () => {
  const joueur = scores(undefined, inscrireJoueur("J1"));
  expect(scores(joueur, voleeChiffree("J1", 15, 1))).toMatchSnapshot();
});

const VOLEE_NULLE = 0;
it("divise en 2 si volée nulle", () => {
  const joueur = scores(undefined, inscrireJoueur("J1"));
  expect(
    scores(joueur, voleeChiffree("J1", 15, VOLEE_NULLE))[0].points
  ).toEqual(20);
});

it("arrondi au point supérieur si besoin", () => {
  let joueur = scores(undefined, inscrireJoueur("J1"));
  joueur[0].points = 25;
  expect(
    scores(joueur, voleeChiffree("J1", 15, VOLEE_NULLE))[0].points
  ).toEqual(13);
});

it("gère le rang des doubles", () => {
  let score = scores(undefined, inscrireJoueur("J1"));
  score = scores(score, voleeChiffree("J1", 15, 1));
  score = scores(score, voleeChiffree("J1", 16, 1));
  score = scores(
    score,
    voleeSurDouble("J1", [
      { chiffre: 20, nombre: 2 },
      { chiffre: 1, nombre: 4 }
    ])
  );

  expect(score[0].points).toBe(40 + 15 + 16 + 2 * 20 + 4 * 1);
});
