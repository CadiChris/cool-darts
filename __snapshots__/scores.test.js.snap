// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`donne des points 1`] = `
Array [
  Object {
    "joueur": "J1",
    "points": 55,
    "touches": Object {
      "15": Array [
        Object {
          "chiffre": 15,
          "nombre": 1,
        },
      ],
    },
  },
]
`;
